app/models/product.rb

```diff
@@ -55,10 +58,18 @@ def discount_price_cannot_be_greater_than_price_value
   end
 
   def split_description_to_words
-    description.split(' ')
+    description.split if description?
   end
 
   def self.human_attribute_name(attr, options = {})
     HUMANIZED_ATTRIBUTES[attr.to_sym] || super
   end
+
+  def ensure_title_has_a_value
+    self.title = 'abc' if self.title.blank?
+  end
+
+  def ensure_default_discount_price
+    self.discount_price = self.price unless self.discount_price?
```

Comment:
let's pass `unless` condition as an option to the callback method.
Also we don't need to use `self` when reading an attribute